### DISCUSSION TITLE
JDBC need to iterate through ResultSet ?

### DISCUSSION THREAD

**Discussion by david99s** [duckdb/duckdb#2387] [Q&A]
Hey, 
i was wondering what is actually happen when executing a SQL Query and storing the results in a ResultSet regarding DuckDB.
Doey the ResultSet already contains all the answer tuples for the query or is going step by step while iterating through the ResultSet. I come up with this question since the performance time needed is really different. 
`long startTime = System.currentTimeMillis();
                Statement b = conn.createStatement();
                ResultSet rs2 = b.executeQuery("Select o_orderkey, o_orderstatus, o_orderdate, o_orderpriority, o_comment from orders");
                while(rs2.next()){
                    int okey=rs2.getInt(1);
                    String st=rs2.getString(2);
                    Date dt=rs2.getDate(3);
                    String pr=rs2.getString(4);
                    String co=rs2.getString(5);
                }
                long endTime = System.currentTimeMillis();
                System.out.println(i+". DuckDB " + (endTime- startTime) +" ms");
                sumD+=(endTime- startTime);`
When i only measure the time for creating the ResultSet wihtout iterating through it, the measured time is just a fraction of it. 
Since DuckDB answers the query by vectorized blocks i was wondering if there are already all the answer tuples inside the ResultSet. 
Thank you!

**Comment by hannes** [comment]
Indeed it's automatically fetching new parts of the result as needed.

**Reply by david99s** [reply]
So i have to iterate through the ResultSet otherwise i don't have the whole result tuples, did i understand that correctly?
